apps/emqx_authn/src/simple_authn/emqx_authn_http.erl

```diff
@@ -285,9 +285,9 @@ parse_url(Url) ->
                     BaseUrl = iolist_to_binary([Scheme, "//", HostPort]),
                     case string:split(Remaining, "?", leading) of
                         [Path, QueryString] ->
-                            {BaseUrl, Path, QueryString};
```

Comment:
Not that it should be done in this PR, but I feel this `parse_url/1` fn could be replaced by `uri_string:parse` :thinking: 

(same for `apps/emqx_authz/src/emqx_authz_http.erl`)

Reply:
Unfortunately not 😅 This is an old conceptual issue — that users provide _templates_ for urls `https://my.site/user/${username}?password=${password}` which are not valid urls and are treated as invalid by URL parsers.

Reply:
Seems that this problem was the reason for appearing of this loose parser.